### ISSUE TITLE
Overdrawn symbols at tile corner

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-js#4617]
If you have a symbol layer with text for a feature at null island, at any zoom level other than 0, it'll be overdrawn. Presumably what's happening is that the feature appears within all of the four tiles which have a corner at null island, and is getting drawn by all of them.

Test case:

```
{
  "version": 8,
  "metadata": {
    "test": {
      "height": 64,
      "width": 64
    }
  },
  "zoom": 1,
  "sources": {
    "point": {
      "type": "geojson",
      "data": {
        "type": "Point",
        "coordinates": [ 0, 0 ]
      }
    }
  },
  "glyphs": "local://glyphs/{fontstack}/{range}.pbf",
  "layers": [
    {
      "id": "text",
      "type": "symbol",
      "source": "point",
      "layout": {
        "text-field": "xxx",
        "text-font": [
          "Open Sans Semibold",
          "Arial Unicode MS Bold"
        ]
      }
    }
  ]
}
```

Expected result:

![image](https://cloud.githubusercontent.com/assets/98601/25204389/2e1f552e-2512-11e7-833d-da3696198ecf.png)

Actual result:

![image](https://cloud.githubusercontent.com/assets/98601/25204414/3e6446ec-2512-11e7-98cf-ba2aa2606ce0.png)

**Comment by jfirebaugh** [comment]
See #4729 for another demo of this issue.

**Comment by jfirebaugh** [comment]
@ChrisLoer This should be fixed by viewport collision detection, right?

**Comment by ChrisLoer** [comment]
@jfirebaugh Yup, I'll turn this into a regression test case and close it.

**Comment by ChrisLoer** [comment]
Fixed in #5150, test in #5515.
